src/modeler/AutoRest.Swagger/Validation/Extensions/LongRunningExtensionRule.cs

```diff
@@ -12,52 +12,39 @@ public abstract class LongRunningExtensionRule : ExtensionRule
     {
         protected override string ExtensionName => "x-ms-long-running-operation";
 
-        protected static bool IsValidResponseCodes(RuleContext context)
-        {
-            Dictionary<string, OperationResponse> responses = context?.GetFirstAncestor<Operation>()?.Responses;
-            var method = context?.Parent?.Parent?.Key;
-
-            if (method == null)
-            {
-                return false;
-            }
-
-            bool isPutPatchResponseCodesValid = true;
-            bool isPostResponseCodesValid = true;
-            bool isDeleteResponseCodesValid = true;
-
-            switch (method.ToLower())
-            {
-                case "put":
-                case "patch":
-                    isPutPatchResponseCodesValid = IsPutPatchResponseCodesValid(responses);
-                    break;
-                case "post":
-                    isPostResponseCodesValid = IsPostResponseCodesValid(responses);
-                    break;
-                case "delete":
-                    isDeleteResponseCodesValid = IsDeleteResponseCodesValid(responses);
-                    break;
-                default:
-                    break;
-            }
-
-            return isPutPatchResponseCodesValid && isPostResponseCodesValid && isDeleteResponseCodesValid;
-        }
+        /// <summary>
+        /// Id of the Rule.
+        /// </summary>
+        public override string Id => "M2005";
 
-        public abstract override string MessageTemplate { get; }
+        /// <summary>
+        /// Violation category of the Rule.
+        /// </summary>
+        public override ValidationCategory ValidationCategory => ValidationCategory.SDKViolation;
 
         /// <summary>
         /// The severity of this message (ie, debug/info/warning/error/fatal, etc)
         /// </summary>
-        public abstract override Category Severity { get; }
+        public override Category Severity => Category.Error;
+
+        /// <summary>
+        /// Returns http verb for the given Rule context.
+        /// </summary>
+        /// <param name="context">Rule context.</param>
+        /// <returns>http verb as string</returns>
+        protected static string GetHttpVerb(RuleContext context)
+        {
+            var method = context?.Parent?.Parent?.Key;
+
+            return method ?? null;
+        }
 
         /// <summary>
         /// Validate that Put or Patch long running operations must have valid terminal status code 200 or 201.
         /// </summary>
         /// <param name="responses">Dictionary of responses for the operation.</param>
         /// <returns><c>true</c> if at least one valid terminal status code availabe, otherwise <c>false</c>.</returns>
-        private static bool IsPutPatchResponseCodesValid(Dictionary<string, OperationResponse> responses)
+        protected static bool IsPutPatchResponseCodesValid(Dictionary<string, OperationResponse> responses)
```

Comment:
While we are at it let's get this straight
The whole function could be replaced by:
```private static bool IsPutPatchResponseCodesValid(Dictionary<string, OperationResponse> responses) => responses?.GetValueOrNull("200")!=null && responses?.GetValueOrNull("201")!=null;```
The method is unnecessarily verbose and quite inexplicably has the same if condition repeated 😮